sklearn/inspection/_h_statistic.py

```diff
@@ -260,17 +268,16 @@ def h_statistic(
     if not (hasattr(X, "__array__") or sparse.issparse(X)):
         X = check_array(X, force_all_finite="allow-nan", dtype=object)
 
-    if sample_weight is not None:
-        sample_weight = _check_sample_weight(sample_weight, X)
+    sample_weight = _check_sample_weight(sample_weight, X)
 
     # Usually, the data is too large and we need subsampling
-    if X.shape[0] > n_max:
-        row_indices = sample_without_replacement(
-            n_population=X.shape[0], n_samples=n_max, random_state=random_state
+    if _num_samples(X) > subsample:
+        X, sample_weight = resample(
```

Comment:
We are using this utility in some other places. This should avoid a bit of boilerplate code.

Reply:
Looks good! I think I have picked some snippets from permutation importance.